ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -0,0 +1,382 @@
+
+    try
+    {
+        meta_data = tue::config::fromFile(filename);
+    }
+    catch (tue::config::ParseException& e)
+    {
+        std::cerr << "Could not open '" << filename << "'.\n\n" << e.what() << std::endl;
+        return false;
+    }
+
+    tue::config::Reader r(meta_data);
+
+    // Read image
+    std::string rgbd_filename;
+    if (r.value("rgbd_filename", rgbd_filename))
+    {
+        tue::filesystem::Path abs_rgbd_filename = tue::filesystem::Path(filename).parentPath().join(rgbd_filename);
+
+        std::ifstream f_rgbd;
+        f_rgbd.open(abs_rgbd_filename.string().c_str(), std::ifstream::binary);
+
+        if (!f_rgbd.is_open())
+        {
+            std::cerr << "Could not open '" << filename << "'." << std::endl;
+            return false;
+        }
+
+        image.reset(new rgbd::Image);
+
+        tue::serialization::InputArchive a_in(f_rgbd);
+        rgbd::deserialize(a_in, *image);
+    }
+
+    // Read sensor pose
+    if (!ed::deserialize(r, "sensor_pose", sensor_pose))
+    {
+        std::cerr << "No field 'sensor_pose' specified." << std::endl;
+        return false;
+    }
+
+//    if (r.hasError())
+//    {
+//        std::cout << "Error while reading file '" << filename << "':\n\n" << r.error() << std::endl;
+//        return false;
+//    }
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+bool loadWorldModel(const std::string& model_name, ed::WorldModel& world_model)
+{
+    ed::UpdateRequest req;
+
+    ed::models::ModelLoader model_loader;
+
+    std::stringstream error;
+    if (!model_loader.create("_root", model_name, req, error, true))
+    {
+        std::cerr << "Model '" << model_name << "' could not be loaded:" << std::endl << std::endl;
+        std::cerr << error.str() << std::endl;
+        return false;
+    }
+
+    // Reset world
+    world_model = ed::WorldModel();
+
+    // Update world
+    world_model.update(req);
+
+    return true;
+}
+
+// ----------------------------------------------------------------------------------------------------
+
+void usage()
+{
+    std::cout << "Usage: ed_segmenter IMAGE-FILE-OR-DIRECTORY  WORLDMODEL_NAME  ENTITY_ID" << std::endl;
+}
+
+// Getting roll, pitch and yaw from a quaternion,
+// copied from https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles
+
+void drawLine( cv::Mat& canvas, geo::Vec2 point1, geo::Vec2 point2, geo::Transform2 pose, float resolution, int origin_x, int origin_y, cv::Scalar color)
+{
+    // computing points relative to the pose (in meters)
+    geo::Vec2 rel_point1 = pose * point1;
+    geo::Vec2 rel_point2 = pose * point2;
+
+    // position to pixels
+    int x_p1 = origin_x + (int)(rel_point1.x * resolution);
+    int y_p1 = origin_y - (int)(rel_point1.y * resolution);
+    int x_p2 = origin_x + (int)(rel_point2.x * resolution);
+    int y_p2 = origin_y - (int)(rel_point2.y * resolution);
```

Comment:
When using explicit casting, use modern style casting, `static_cast`/`dynamic_cast`, not the old c-style casting.

P.S. double to int casting is truncating it, so behaviour of `floor`. Is this the correct behaviour?